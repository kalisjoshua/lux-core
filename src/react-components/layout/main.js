/** @memberof module:react-components/Layout */

import React from 'react'; // Even if this isn't used it needs to be imported!

function mainComponent() {

  return (
    <main className="body-divider pbl" role="main">
      {/* <!-- page content --> */}
      <p>Default Page Render</p>
    </main>
  );
}
mainComponent.propTypes = {
};

export default mainComponent;
