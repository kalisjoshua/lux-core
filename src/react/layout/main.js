/**
 * @module Main
 * @memberof lux-react
 */

import React from 'react'; // Even if this isn't used it needs to be imported!

import registry from '../componentRegistry';
import shapeOfSiren from '../siren.react';

function childFinder() {}

function mainComponent(props) {

  return (
    <main className="body-divider pbl" id="maincontent" role="main">
      {childFinder(props.class)}
    </main>
  );
}
mainComponent.propTypes = {
  ...shapeOfSiren
};

registry('Main', mainComponent);
