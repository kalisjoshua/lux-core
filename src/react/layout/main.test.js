import React from 'react';
import renderer from 'react-test-renderer';

import registry from '../componentRegistry';

import './main';
const Main = registry('Main');

describe('Main (supplied Layout)', function () {
  it('should match the snapshot', function () {
    const component = renderer.create(
      <Main />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
