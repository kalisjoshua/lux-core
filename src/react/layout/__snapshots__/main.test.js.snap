exports[`Main (supplied Layout) should match the snapshot 1`] = `
<main
  className="body-divider pbl"
  id="maincontent"
  role="main" />
`;
